Core/src/Seeding2/DoubletSeedFinder.cpp

```diff
@@ -113,10 +103,15 @@ void createDoubletsImpl(
   }
 
   for (SpacePointIndex2 otherSpIndex : candidateSps.subspan(candidateOffset)) {
-    ConstSpacePointProxy2 otherSp = spacePoints.at(otherSpIndex);
+    ConstSpacePointProxy2 otherSp = spacePoints[otherSpIndex];
+
+    const float xO = otherSp.x();
+    const float yO = otherSp.y();
+    const float zO = otherSp.z();
+    const float rO = otherSp.r();
```

Comment:
This I'm not 100% sure. When I tried this in the previous Seeding code, retrieving the members of the spacepoints was quite slow so that's why the code was doing it only if they were passing the cuts. 
Did you have a chance to see if this helps?

Reply:
I tried to see if this does anything about performance but for what I measured in ACTS this has 0 impact. In principle we can revert this but I thought it makes the data flow a bit more visible